### ISSUE TITLE
[CMake Cache Editor] drop-down lists are not working properly

### ISSUE THREAD

**Issue by tehKaiN** [microsoft/vscode-cmake-tools#3117]
### Brief Issue Summary

Consider following code (from https://github.com/AmigaPorts/ACE):

```cmake
set(ACE_LIBRARY_KIND OBJECT CACHE STRING "Build kind: static library or bunch of .o files. OBJECT allows for more link-time optimizations.")
set(ACE_LIBRARY_KINDS "OBJECT" "STATIC")
set_property(CACHE ACE_LIBRARY_KIND PROPERTY STRINGS ${ACE_LIBRARY_KINDS})
```

This results in following UI, which is not correct:

![obraz](https://user-images.githubusercontent.com/16180344/232697000-2521abd9-3d9c-494d-b47a-33880aa85eb3.png)

however, clicking to expand the drop-down list does the following:

![obraz](https://user-images.githubusercontent.com/16180344/232697125-bf696b85-45a8-4103-bc7d-c6b291fc6ec8.png)

- the drop-down list is in wrong position
- other options than currently cached don't show up

### CMake Tools Diagnostics

```shell
n/a
```

### Debug Log

```shell
n/a
```

**Comment by benmcmorran** [comment]
Thanks for the report! I can reproduce this and we'll try to get a fix out soon.

**Comment by gcampbell-msft** [comment]
@tehKaiN A quick update.

The way that the list works is that it filters the search results based on what you've entered. We are using the `datalist` html element, which filters. Thus, if you want to see all of the available options, you should remove any text input. This should unblock you in regards to selecting any other options. 

I'm still investigating why it appears in a weird location.

**Comment by tehKaiN** [comment]
Indeed, it works! Dunno if that's good though, since for me it works kinda counter-intuitive, but maybe it's just me. Ideally, if that's possible and isn't *that* complex, the drop-down button could show all the options, whereas while typing there could be a second drop-down which only shows suggestions - I think that's usual native way of how combo boxes work.

Also currently, when input is empty, I can't press the down arrow to cycle through values - it shows the list, but doesn't select anything on consecutive button presses - this is quite different behavior from bare datalist html element, isn't it? Perhaps some script is playing tricks there.

**Comment by gcampbell-msft** [comment]
@tehKaiN We've narrowed this down to a likely bug in VSCode itself and have tracked it here https://github.com/microsoft/vscode/issues/180253.

Thanks!

**Comment by gcampbell-msft** [comment]
Tracking here: https://github.com/microsoft/vscode/issues/129999
